airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source_adapter.py

```diff
@@ -58,3 +68,60 @@ def _select_abstract_streams(self, config: Mapping[str, Any], configured_catalog
             if isinstance(stream_instance, AbstractStreamFacade):
                 abstract_streams.append(stream_instance.get_underlying_stream())
         return abstract_streams
+
+    def _convert_to_concurrent_stream(self, logger: logging.Logger, stream: Stream, cursor: Optional[Cursor] = None) -> Stream:
+        """
+        Prepares a stream for concurrent processing by initializing or assigning a cursor,
+        managing the stream's state, and returning an updated Stream instance.
+        """
+        state: MutableMapping[str, Any] = {}
+
+        if cursor:
+            state = cursor.state
+
+            stream.cursor = cursor  # type: ignore[assignment]  # cursor is of type ConcurrentCursor, which inherits from Cursor
+            if hasattr(stream, "parent"):
+                stream.parent.cursor = cursor
+        else:
+            cursor = FinalStateCursor(
+                stream_name=stream.name,
+                stream_namespace=stream.namespace,
+                message_repository=self.message_repository,  # type: ignore[arg-type]  # _default_message_repository will be returned in the worst case
+            )
+        return StreamFacade.create_from_stream(stream, self, logger, state, cursor)
+
+    def initialize_cursor(
+        self,
+        stream: Stream,
+        state_manager: ConnectorStateManager,
+        converter: AbstractStreamStateConverter,
+        slice_boundary_fields: Optional[Tuple[str, str]],
+        start: Optional[CursorValueType],
+        end_provider: Callable[[], CursorValueType],
+        lookback_window: Optional[GapType] = None,
+        slice_range: Optional[GapType] = None,
+    ) -> Optional[ConcurrentCursor]:
+        lookback_window = timedelta(seconds=DEFAULT_LOOKBACK_SECONDS) if lookback_window is None else lookback_window
+
+        cursor_field_name = stream.cursor_field
+
+        if cursor_field_name:
+            stream_state = state_manager.get_stream_state(stream.name, stream.namespace)
+            cursor_field = CursorField(cursor_field_name) if isinstance(cursor_field_name, str) else CursorField(cursor_field_name[0])
```

Comment:
I think I understand why you coded it this way since Stripe was previously doing this. However, it might be a bit aggressive, but I think that we should fail this stream if we're using a composite cursor field name. It's more transparent and could lead to unexpected behavior if we truncate this.

It also is more consistent with some of our other concurrent CDK code in `streams/concurrent/helpers.py` where we raise a `ValueError` when there is more than one field.